streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionTupleForwarder.java

```diff
@@ -46,14 +46,14 @@
         cachingEnabled = ((WrappedStateStore) store).setFlushListener(flushListener, sendOldValues);
     }
 
-    public void maybeForward(final Windowed<K> key,
-                             final V newValue,
-                             final V oldValue) {
+    public void maybeForward(final Record<Windowed<K>, Change<V>> record) {
         if (!cachingEnabled) {
-            context.forward(new Record<>(
-                key,
-                new Change<>(newValue, sendOldValues ? oldValue : null),
-                key.window().end()));
+            context.forward(
+                record.withValue(
+                        new Change<>(
+                            record.value().newValue,
+                            sendOldValues ? record.value().oldValue : null))
```

Comment:
Huh, does it make sense for the forwarder to check sendOldValues? It seems like the pattern for forwarders is to push that check into the Processor. Then again, it may not be worth messing with it in this PR.